### PR TITLE
refactor(core, docs, style): Rename project to simple-resume, refacto…

### PR DESCRIPTION
**Pull request by athola**
…r API, and update docs

- Renamed `easyresume` to `simple-resume` across the codebase.
- Introduced a new functional API for resume generation and session management.
- Added comprehensive documentation including usage guides, migration steps, and architectural decision records.
- Refactored CSS architecture for better separation of styles.
- Improved SVG icon contrast rules for better rendering.

BREAKING CHANGE:
- Project renamed from `easyresume` to `simple-resume`.
- Programmatic API has changed, requiring migration for existing users.
- Template structure for dictionary access has changed from dot notation to bracket notation.

### CODE REVIEW

File: action_plan.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+### 30-60-90 Day Action Plan for Homeward
```
Comment (athola) [inline action_plan.txt]: move out of this repo

File: src/simple_resume/core/resume.py [hunk 2]
```diff
@@ -0,0 +1,1374 @@
...
+
+        Returns:
+            GenerationResult with metadata and operations
+
+        Raises:
+            ValueError: If format is not supported
+
+        """
+        format = format.lower()
+        if format == "pdf":
```
Comment (athola) [inline src/simple_resume/core/resume.py]: make these Enums

File: src/simple_resume/core/resume.py [hunk 3]
```diff
@@ -0,0 +1,1374 @@
...
+            self._render_plan = self.prepare_render_data(
+                source_data, preview=actual_preview, base_path=base_path
+            )
+        return self._render_plan
+
+    def _generate_pdf(
+        self, render_plan: RenderPlan, output_path: Path, **kwargs: Any
+    ) -> GenerationResult | tuple[GenerationResult, int | None]:
+        """Dispatch PDF generation to the appropriate backend."""
+        if render_plan.mode == "latex":
```
Comment (athola) [inline src/simple_resume/core/resume.py]: the mode should be Enum

File: src/simple_resume/core/resume.py [hunk 4]
```diff
@@ -0,0 +1,1374 @@
...
+        source_yaml_content: dict[str, Any], mode: RenderMode
+    ) -> dict[str, Any]:
+        """Transform YAML content based on render mode."""
+        if mode == "latex":
+            return copy.deepcopy(source_yaml_content)
+
+        return render_markdown_content(source_yaml_content)
+
+    @staticmethod
+    def _build_render_plan(  # noqa: PLR0913
```
Comment (athola) [inline src/simple_resume/core/resume.py]: remove this noqa and address it

File: PACKAGING.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# Packaging Simple-Resume for PyPI
+
+This guide outlines the process for preparing, building, and publishing Simple-Resume releases to PyPI.
+
+## Prerequisites
+
+- Python 3.12
```
Comment (athola) [inline PACKAGING.md]: Make a mention that 3.9+ instead since we should be supporting 3.9 and above

File: pyproject.toml [hunk 6]
```diff
@@ -167,3 +182,10 @@ ignore_missing_imports = true
 [[tool.mypy.overrides]]
 module = "tests.*"
 disallow_untyped_decorators = false
+disallow_untyped_defs = false
```
Comment (athola) [inline pyproject.toml]: Remove these

File: pyproject.toml [hunk 7]
```diff
@@ -126,6 +135,12 @@ extend-ignore = [
     "RUF002",   # Allow unicode characters in docstrings
     "RUF003",   # Allow unicode characters in comments
     "S101",     # Allow assert statements in tests
+    "D100",     # Relax docstring requirements for test modules
```
Comment (athola) [inline pyproject.toml]: Remove these

File: src/simple_resume/latex_renderer.py [hunk 8]
```diff
@@ -0,0 +1,644 @@
...
+        )
+
+    pdf_path = tex_path.with_suffix(".pdf")
+    return pdf_path
+
+
+def compile_tex_to_html(
+    tex_path: Path,
+    *,
+    tools: Iterable[str] = ("pandoc", "htlatex"),
```
Comment (athola) [inline src/simple_resume/latex_renderer.py]: Make these enum

File: src/simple_resume/palettes/generators.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+    def __init__(self, seed: int):
+        self.seed = seed
+        self.state = seed
+
+    def random(self) -> float:
+        """Generate deterministic random float between 0 and 1."""
+        self.state += 1
+        hash_input = f"{self.seed}-{self.state}".encode()
+        hash_bytes = hashlib.sha256(hash_input).digest()
```
Comment (athola) [inline src/simple_resume/palettes/generators.py]: This doesn't appear efficient, we're just taking a slice of the first 8 bytes

File: src/simple_resume/palettes/registry.py [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+def _load_palettable(registry: PaletteRegistry) -> None:
+    """Populate registry with palettable palettes."""
+    for record in ensure_bundled_palettes_loaded():
+        palette = load_palettable_palette(record)
+        if palette is not None:
+            registry.register(palette)
+
+
+@lru_cache(maxsize=1)
+def get_global_registry() -> PaletteRegistry:
```
Comment (athola) [inline src/simple_resume/palettes/registry.py]: Call this get_palette_registry

File: src/simple_resume/shell/generation.py [hunk 11]
```diff
@@ -0,0 +1,578 @@
...
+        # Load raw data (I/O)
+        raw_data = get_content(name, paths=paths, transform_markdown=False)
+
+        # Create render plan using core logic (pure)
+        plan = Resume.prepare_render_data(
+            source_yaml_content=raw_data, preview=False, base_path=str(paths.content)
+        )
+
+        # Execute the plan (I/O)
+        if plan.mode == "latex":
```
Comment (athola) [inline src/simple_resume/shell/generation.py]: should compare enum

File: src/simple_resume/shell/generation.py [hunk 12]
```diff
@@ -0,0 +1,578 @@
...
+        # Load raw data (I/O)
+        raw_data = get_content(name, paths=paths, transform_markdown=False)
+
+        # Create render plan using core logic (pure)
+        plan = Resume.prepare_render_data(
+            source_yaml_content=raw_data, preview=True, base_path=str(paths.content)
+        )
+
+        # Execute the plan (I/O)
+        if plan.mode == "latex":
```
Comment (athola) [inline src/simple_resume/shell/generation.py]: should compare enum

File: src/simple_resume/utilities.py [hunk 13]
```diff
@@ -0,0 +1,761 @@
...
+    """Calculate appropriate text color for given background color.
+
+    Args:
+        background_color: Background color as hex string (e.g., "#FF0000")
+
+    Returns:
+        Text color as hex string that provides good contrast
+
+    """
+    return _get_contrasting_text_color(background_color)
```
Comment (athola) [inline src/simple_resume/utilities.py]: why are we exposing a public wrapper for this private method?

File: tests/unit/test_core_resume.py [hunk 14]
```diff
@@ -0,0 +1,1247 @@
+"""Unit tests for core resume transformations - pure functions without side effects.
+
```
Comment (athola) [inline tests/unit/test_core_resume.py]: split this up as well when we split up `src/simple-resume/resume.py`

File: tests/unit/test_generation.py [hunk 15]
```diff
@@ -0,0 +1,544 @@
...
+class TestGenerateAll:
+    """Test the generate_all function."""
+
+    def test_generate_all_invalid_formats(self, story):
+        story.given("an unsupported format is included")
+        config = GenerationConfig(formats=["pdf", "docx"])
+        with pytest.raises(ValueError, match="Unsupported format"):
+            generate_all(config)
+
+    @patch("simple_resume.generation.validate_directory_path")
```
Comment (athola) [inline tests/unit/test_generation.py]: why is this patched twice?

File: tests/unit/test_shell_generation.py [hunk 16]
```diff
@@ -0,0 +1,982 @@
+"""Unit tests for shell generation orchestration with I/O dependencies.
```
Comment (athola) [inline tests/unit/test_shell_generation.py]: this file should do a better job of using the `story` test utility

File: tests/unit/test_template_resolution_fix.py [hunk 17]
```diff
@@ -0,0 +1,718 @@
...
+                            "weasyprint.HTML was not passed HTML content"
+                        )
+                    assert "Test User" in rendered_html
+
+                    mock_css_class.assert_called_once()
+                    css_args, css_kwargs = mock_css_class.call_args
+                    css_string = css_kwargs.get("string") or (
+                        css_args[0] if css_args else ""
+                    )
+                    assert "210mm 297mm" in css_string
```
Comment (athola) [inline tests/unit/test_template_resolution_fix.py]: this should be set from a variable in case it may change in the future

File: tests/unit/test_utilities.py [hunk 18]
```diff
@@ -1,17 +1,4 @@
...
-3. REFACTOR: Improve code while keeping tests green
-
-Tests follow extreme programming practices:
-- Edge case coverage
-- Business logic validation
-- Clear, descriptive test names
-- Single assertion per test where possible
-- Test isolation and independence
-"""
+"""Behavioural tests for simple_resume.utilities using BDD terminology."""
```
Comment (athola) [inline tests/unit/test_utilities.py]: this file should use the `story` test utilitiy

File: wiki/Contributing.md [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+To contribute code:
+
+1. Fork and clone the repository.
+2. Create a new branch for your changes.
+3. Set up your development environment using the [Development Guide](Development-Guide.md).
+4. Make your changes, add tests, and update documentation as needed.
+5. Run all checks:
+
+    ```bash
+    make check-all
```
Comment (athola) [inline wiki/Contributing.md]: add `make validate` here too

File: wiki/Migration-Guide.md [hunk 20]
```diff
@@ -0,0 +1,381 @@
...
+
+**Before (v0.0.x):**
+
+```bash
+# Old direct Python file execution
+uv run python src/simple_resume/generate_pdf.py --data-dir resume_private
+uv run python src/simple_resume/generate_html.py --open
+```
+
+**After (v0.2.0+):**
```
Comment (athola) [inline wiki/Migration-Guide.md]: we are on version 0.1.0

File: wiki/Migration-Guide.md [hunk 21]
```diff
@@ -0,0 +1,381 @@
...
+**Before (v0.0.x):**
+
+```python
+from simple_resume.generation import generate_pdf as generate_pdf_cli
+
+# Old approach: Call CLI function directly
+generate_pdf_cli()  # programmatic entry point now exposed via generation module
+```
+
+**After (v0.2.0+):**
```
Comment (athola) [inline wiki/Migration-Guide.md]: we are on version 0.1.0

File: wiki/Migration-Guide.md [hunk 22]
```diff
@@ -0,0 +1,381 @@
...
+```python
+# Old CLI-style import
+from simple_resume.generation import generate_pdf as generate_resume_pdf
+from simple_resume.utilities import load_yaml_data
+
+data = load_yaml_data("resume.yaml")
+pdf_path = generate_resume_pdf(data, "output.pdf")
+```
+
+**After (v0.2.0+):**
```
Comment (athola) [inline wiki/Migration-Guide.md]: we are on version 0.1.0

File: wiki/Migration-Guide.md [hunk 23]
```diff
@@ -0,0 +1,381 @@
...
+Simple-ResumeError (base)
+├── ResumeValidationError    # Invalid YAML structure
+├── TemplateRenderError      # Jinja2 rendering issues
+├── FileOperationError       # File not found, permission denied
+└── PaletteError             # Color scheme issues
+```
+
+## New Features
+
+The v0.2.0+ release introduces these new features:
```
Comment (athola) [inline wiki/Migration-Guide.md]: we are on version 0.1.0

File: wiki/Technical-Debt-Resolution.md [hunk 24]
```diff
@@ -1,55 +1,54 @@
 # Technical Debt Resolution
```
Comment (athola) [inline wiki/Technical-Debt-Resolution.md]: let's remove this

File: README.md [hunk 25]
```diff
@@ -1,108 +1,237 @@
...
-## Command Line Interface
+# Render multiple formats at once (name inferred from YAML path)
+results = generate("resume_private/input/my_resume.yaml", formats=("pdf", "html"))
+
+# Launch an HTML preview in your browser
+preview("resume_private/input/my_resume.yaml", open_after=True)
+```
 
-### generate-pdf
+### 3. Use Custom Styling
```
[the remark this answers is not in the document]
Comment (athola) [reply README.md]: done

File: bandit-report.json [hunk 26]
```diff
@@ -0,0 +1,342 @@
+{
```
[the remark this answers is not in the document]
Comment (athola) [reply bandit-report.json]: done

File: src/simple_resume/core/resume.py (hunk 2 above)
Comment (athola) [reply src/simple_resume/core/resume.py]: done

File: src/simple_resume/core/resume.py (hunk 3 above)
Comment (athola) [reply src/simple_resume/core/resume.py]: addressed

File: PACKAGING.md (hunk 5 above)
Comment (athola) [reply PACKAGING.md]: addressed

File: README.md [hunk 27]
```diff
@@ -1,108 +1,237 @@
...
-All project guides are stored in the repo [wiki](wiki/) (mirroring the GitHub wiki):
+## Getting Started
 
-- `wiki/Markdown-Guide.md` – author Markdown-rich Resume content.
-- `wiki/Color-Schemes.md` – customize colors with preset themes or create your own.
-- `wiki/Workflows.md` – understand the CI/CD pipeline and quality gates.
+### Prerequisites
 
-## Installation
+- Python 3.10 or higher
```
[the remark this answers is not in the document]
Comment (athola) [reply README.md]: addressed

File: pyproject.toml (hunk 6 above)
Comment (athola) [reply pyproject.toml]: addressed

File: pyproject.toml (hunk 7 above)
Comment (athola) [reply pyproject.toml]: addressed

File: src/simple_resume/generation.py [hunk 28]
```diff
@@ -0,0 +1,520 @@
...
+        if source_path.suffix.lower() in _YAML_SUFFIXES:
+            return source_path.parent, source_path.stem
+
+    raise ValueError(
+        "Unable to infer data_dir from source. Provide a YAML path, directory, "
+        "or set data_dir explicitly."
+    )
+
+
+def generate(  # noqa: PLR0913
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/generation.py]: addressed

File: src/simple_resume/latex_renderer.py [hunk 29]
```diff
@@ -0,0 +1,644 @@
...
+    if not description:
+        return []
+
+    lines = description.strip("\n").splitlines()
+    blocks: list[Block] = []
+    current_items: list[str] = []
+    ordered = False
+
+    def flush_items() -> None:
+        nonlocal current_items, ordered
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/latex_renderer.py]: Enables clean state management in the nested function. Avoids complex parameter passing and return value unpacking. Maintains encapsulation, as state is contained within the function scope. Supports the parsing algorithm that needs to accumulate items across multiple iterations. And finally, provides readable, maintainable code that clearly expresses the original intent

File: src/simple_resume/palettes/registry.py [hunk 30]
```diff
@@ -0,0 +1,78 @@
...
+    """Return a singleton registry populated with known sources."""
+    registry = PaletteRegistry()
+    for palette in load_default_palettes():
+        registry.register(palette)
+    _load_palettable(registry)
+    return registry
+
+
+def reset_global_registry() -> None:
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/palettes/registry.py]: done

File: src/simple_resume/shell/generation.py (hunk 12 above)
Comment (athola) [reply src/simple_resume/shell/generation.py]: addressed

File: src/simple_resume/utilities.py [hunk 31]
```diff
@@ -0,0 +1,761 @@
...
+            else ((c_val + LINEARIZATION_OFFSET) / (1 + LINEARIZATION_OFFSET))
+            ** LINEARIZATION_EXPONENT
+        )
+
+    r_linear = _linearize(r)
+    g_linear = _linearize(g)
+    b_linear = _linearize(b)
+
+    # Calculate luminance using WCAG weights
+    return 0.2126 * r_linear + 0.7152 * g_linear + 0.0722 * b_linear
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/utilities.py]: Added an inline explanation

File: tests/unit/test_generation.py (hunk 15 above)
Comment (athola) [reply tests/unit/test_generation.py]: The duplicate `@patch("simple_resume.generation.validate_directory_path")` decorators are there because `generate_all` calls `validate_directory_path` twice internally; once for the input/data directory and once for the output directory

File: tests/unit/test_utilities.py (hunk 18 above)
Comment (athola) [reply tests/unit/test_utilities.py]: done

File: wiki/Migration-Guide.md [hunk 32]
```diff
@@ -0,0 +1,381 @@
...
+
+**Before (v0.0.x):**
+
+```jinja2
+{% for item in group.items %}
+  <li>{{ item.title }}: {{ item.description }}</li>
+{% endfor %}
+```
+
+**After (v0.2.0+):**
```
[the remark this answers is not in the document]
Comment (athola) [reply wiki/Migration-Guide.md]: addressed

File: wiki/Migration-Guide.md [hunk 33]
```diff
@@ -0,0 +1,381 @@
...
+**Before (v0.0.x):**
+
+```python
+try:
+    generate_pdf_cli()  # programmatic entry point now exposed via generation module
+except Exception as e:
+    print(f"Error: {e}")
+```
+
+**After (v0.2.0+):**
```
[the remark this answers is not in the document]
Comment (athola) [reply wiki/Migration-Guide.md]: addressed

File: pyproject.toml [hunk 34]
```diff
@@ -135,7 +151,7 @@ skip-magic-trailing-comma = false
 line-ending = "auto"
 
 [tool.mypy]
-python_version = "3.10"
+python_version = "3.12"
```
Comment (athola) [inline pyproject.toml]: should this be min 3.9 instead since we support 3.9?

File: src/simple_resume/cli.py [hunk 35]
```diff
@@ -0,0 +1,791 @@
...
+
+    # Show user-friendly message.
+    print(f"{error_type}: {exc}")
+    if suggestion:
+        print(f"Suggestion: {suggestion}")
+
+    return exit_code
+
+
+generate_resume = _generate_resume
```
Comment (athola) [inline src/simple_resume/cli.py]: Why are we aliasing this?

File: src/simple_resume/core/color_utils.py [hunk 36]
```diff
@@ -0,0 +1,127 @@
...
+            value / LINEARIZATION_DIVISOR
+            if value <= LINEARIZATION_THRESHOLD
+            else ((value + LINEARIZATION_OFFSET) / (1 + LINEARIZATION_OFFSET))
+            ** LINEARIZATION_EXPONENT
+        )
+
+    r_linear = _linearize(r)
+    g_linear = _linearize(g)
+    b_linear = _linearize(b)
+    return 0.2126 * r_linear + 0.7152 * g_linear + 0.0722 * b_linear
```
Comment (athola) [inline src/simple_resume/core/color_utils.py]: Still think this needs a comment

File: src/simple_resume/shell/generation.py [hunk 37]
```diff
@@ -0,0 +1,578 @@
...
+        return f"{base_tag}\n{html}"
+
+    def _cleanup_latex_artifacts(self, tex_path: Path) -> None:
+        """Remove auxiliary files generated during LaTeX compilation."""
+        for suffix in (".aux", ".log", ".out"):
+            candidate = tex_path.with_suffix(suffix)
+            try:
+                if candidate.exists():
+                    candidate.unlink()
+            except OSError:
```
Comment (athola) [inline src/simple_resume/shell/generation.py]: report unlink failure as warning

**Review by athola — COMMENTED** [review]
address all comments before merging

File: src/simple_resume/cli.py (hunk 35 above)
Comment (athola) [reply src/simple_resume/cli.py]: addressed

File: src/simple_resume/cli.py [hunk 38]
```diff
@@ -0,0 +1,791 @@
...
+        return [str(errors)]
+    return fallback
+
+
+def _validate_single_resume_cli(name: str, data_dir: Path | None) -> int:
+    paths = resolve_paths(data_dir=data_dir) if data_dir else None
+    resume = Resume.read_yaml(name, paths=paths)
+    validation = resume.validate()
+    try:
+        resume.validate_or_raise()
```
[the remark this answers is not in the document]
Comment (athola) [reply src/simple_resume/cli.py]: addressed

**Review by athola — COMMENTED** [review]
address all comments before merging
